app/scripts/components/common/mapbox/projection-selector.tsx

```diff
@@ -0,0 +1,299 @@
+import React from 'react';
+import styled from 'styled-components';
+import mapboxgl from 'mapbox-gl';
+import { CollecticonGlobe } from '@devseed-ui/collecticons';
+import {
+  Dropdown,
+  DropMenu,
+  DropMenuItem,
+  DropTitle
+} from '@devseed-ui/dropdown';
+import { Button, createButtonStyles } from '@devseed-ui/button';
+import { glsp } from '@devseed-ui/theme-provider';
+import { FormFieldsetHeader, FormLegend } from '@devseed-ui/form';
+import { ShadowScrollbar } from '@devseed-ui/shadow-scrollbar';
+
+import { FormFieldsetBodyColumns, FormFieldsetCompact } from '$styles/fieldset';
+import StressedFormGroupInput from '../stressed-form-group-input';
+import { validateRangeNum } from '$utils/utils';
+
+import { useState } from 'react';
+
+const ProjectionOptions = styled.div`
+  padding: ${glsp(0, 1)};
+
+  ${FormFieldsetHeader} {
+    padding-top: ${glsp(0.5)};
+    padding-bottom: 0;
+    border: none;
+  }
+
+  ${FormFieldsetBodyColumns} {
+    padding-top: ${glsp(0.5)};
+    padding-bottom: ${glsp(0.5)};
+  }
+`;
+
+/**
+ * Override Dropdown styles to play well with the shadow scrollbar.
+ */
+const DropdownWithScroll = styled(Dropdown)`
+  padding: 0;
+
+  ${DropTitle} {
+    margin: 0;
+    padding: ${glsp(1, 1, 0, 1)};
+  }
+
+  ${DropMenu} {
+    margin: 0;
+  }
+`;
+
+// Why you ask? Very well:
+// Mapbox's css has an instruction that sets the hover color for buttons to
+// near black. The only way to override it is to increase the specificity and
+// we need the button functions to get the correct color.
+// The infamous instruction:
+// .mapboxgl-ctrl button:not(:disabled):hover {
+//   background-color: rgba(0, 0, 0, 0.05);
+// }
+const SelectorButton = styled(Button)`
+  &&& {
+    ${createButtonStyles({ variation: 'primary-fill', fitting: 'skinny' })}
+  }
+`;
+
+export type Projection = Exclude<
```

Comment:
kind of nitpick It might be beneficial to give more specific name to projection interface? It follows the same pattern with the component which can be confusing.

Reply:
Not sure I understand "follows the same pattern with the component". Can you clarify?

Would you have a naming suggestion?

Reply:
I meant the same naming pattern (PascalCase). I was scamming typescript interface naming recommendation and it seems recommended to think about [why this interface exists](https://google.github.io/styleguide/tsguide.html#naming-style) . so maybe `ProjectionOptions`?

Reply:
Ok, that makes sense!